examples/hosts/app-integration/schema-upgrade/src/migrator.ts

```diff
@@ -127,13 +128,19 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
             // Again, it could be the case that someone else finished the migration during our attach.
             if (migratable.getMigrationState() === MigrationState.migrated) {
                 this._migrationP = undefined;
+                migratedModel.close();
                 this.takeAppropriateActionForCurrentMigratable();
                 return;
             }
 
             // TODO: Support retry
             await migratable.finalizeMigration(containerId);
 
+            // If someone else finalized the migration before us, we should close the one we created.
+            if (migratable.newContainerId !== containerId) {
+                migratedModel.close();
```

Comment:
Should we call `this.takeAppropriateActionForCurrentMigratable()` here as well and return?